server/controllers/request.js

```diff
@@ -0,0 +1,97 @@
+const Request = require("../models/Request");
+const asyncHandler = require("express-async-handler");
+const mongoose = require("mongoose");
+
+// @route GET /request/
+// @desc List of requests for logged in user
+// @access Private
+exports.getRequests = asyncHandler(async (req, res, next) => {
+  const userId = req.user.id;
+  const ownerRequests = await Request.find({ owner: userId });
+  const sitterRequests = await Request.find({ sitter: userId });
+
+  if (!ownerRequests.length && !sitterRequests.length) {
+    res.status(404);
+    throw new Error("No requests found");
+  }
+  res.status(200).json({
+    requestsSend: ownerRequests,
+    requestsReceived: sitterRequests,
+  });
+});
+
+// @route POST /request/
+// @desc Create a new request
+// @access Private
+exports.createRequest = asyncHandler(async (req, res, next) => {
+  const ownerId = req.user.id;
+  const { sitterId, startDate, endDate, serviceType, totalPrice } = req.body;
+
+  const ownerExists = await User.findById(ownerId);
+  const sitterExists = await User.findById(sitterId);
+
+  if (!sitterId || !startDate || !endDate) {
+    res.status(400);
+    throw new Error("One of the required fields hasn't been completed");
+  }
+
+  if (ownerExists && sitterExists && ownerId !== sitterId) {
+    const request = new Request({
+      owner: ownerId,
+      sitter: sitterId,
+      startDate,
+      endDate,
+      serviceType,
+      totalPrice,
+    });
+
+    await request.save();
+    res.status(201).json({ request });
+  }
+
+  res.status(500);
+  throw new Error("Error creating request");
```

Comment:
An alternative way to do this is by `throw`ing the error you want - because we have generic error handler wired via the `error` middleware (sends a `500` code response with the message of the error thrown). Just wanted to mention it 👀 :D